--- /tmp/mkinitcpio.conf	2020-02-01 15:52:43.173336836 +0300
+++ /etc/mkinitcpio.conf	2020-02-01 12:38:42.150015781 +0300
@@ -4,7 +4,7 @@
 # run.  Advanced users may wish to specify all system modules
 # in this array.  For instance:
 #     MODULES=(piix ide_disk reiserfs)
-MODULES=()
+MODULES=(dm_mod)
 
 # BINARIES
 # This setting includes any additional binaries a given user may
@@ -49,7 +49,7 @@
 #
 ##   NOTE: If you have /usr on a separate partition, you MUST include the
 #    usr, fsck and shutdown hooks.
-HOOKS=(base udev autodetect modconf block filesystems keyboard fsck)
+HOOKS=(base udev autodetect modconf block lvm2 filesystems keyboard fsck)
 
 # COMPRESSION
 # Use this to compress the initramfs image. By default, gzip compression
